File: src/java/org/javaturk/dp/ch02/elevator/responsible/ResponsibleElevatorTest.java
package org.javaturk.dp.ch02.elevator.responsible;


public class ResponsibleElevatorTest {
	public static void main(String[] args) {

		Elevator elevator = new ResponsibleElevator();
		elevator.openDoor();  // Get into the elevator
		elevator.go(Floor.FIFTH);

		elevator.openDoor(); // Somebody is coming :)
		elevator.go(Floor.FIRST);
		elevator.go(Floor.FIFTH);
	}
}

File: src/c#/ch02/Elevator/Responsible/ResponsibleElevatorTest.cs
using System;
namespace Elevator.Responsible
{
    public class ResponsibleElevatorTest
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("ResponsibleElevatorTest");

            IElevator elevator = new ResponsibleElevator();
            elevator.OpenDoor();  // Get into the elevator
            elevator.Go(Floor.FIFTH);

            elevator.OpenDoor(); // Somebody is coming :)
            elevator.Go(Floor.FIRST);
            elevator.Go(Floor.FIFTH);
        }
    }
}
